BitList.cs

```diff
@@ -0,0 +1,1552 @@
+            return new ushort[0];
+        else
+        {
+            var shorts = ToInt16Array();
+            var ushorts = new List<ushort>();
+            foreach (var bits in shorts)
+                ushorts.Add((ushort)bits);
+            return ushorts.ToArray();
+        }
+    }
+
+    public uint[] ToUInt32Array()
+    {
+        if (IsEmpty)
+            return new uint[0];
+        else
+        {
+            var ints = ToInt32Array();
+            var uints = new List<uint>();
+            foreach (var bits in ints)
+                uints.Add((uint)bits);
+            return uints.ToArray();
+        }
+    }
+
+    public ulong[] ToUInt64Array()
+    {
+        if (IsEmpty)
+            return new ulong[0];
+        else
+        {
+            var longs = ToInt64Array();
+            var ulongs = new List<ulong>();
+            foreach (var bits in longs)
+                ulongs.Add((ulong)bits);
+            return ulongs.ToArray();
+        }
+    }
+
+    public float[] ToSingleArray()
+    {
+        if (IsEmpty)
+            return new float[0];
+        else
+        {
+            var bytes = ToByteArray();
+            var bytesList = new List<byte[]>();
+            var c = -1;
+            for (int i = 0; i < bytes.Length; i++)
+            {
+                c++;
+                if (i == 0 | c == 4)
+                {
+                    bytesList.Add(new byte[4]);
+                    c = 0;
+                }
+                bytesList[bytesList.Count - 1][c] = bytes[i];
+            }
+            var floatList = new List<float>();
+            foreach (var fbytes in bytesList)
+                floatList.Add(BitConverter.ToSingle(fbytes, 0));
+            return floatList.ToArray();
+        }
+    }
+
+    public double[] ToDoubleArray()
+    {
+        if (IsEmpty)
+            return new double[0];
+        else
+        {
+            var bytes = ToByteArray();
+            var bytesList = new List<byte[]>();
+            var c = -1;
+            for (int i = 0; i < bytes.Length; i++)
+            {
+                c++;
+                if (i == 0 | c == 8)
+                {
+                    bytesList.Add(new byte[8]);
+                    c = 0;
+                }
+                bytesList[bytesList.Count - 1][c] = bytes[i];
+            }
+            var doubleList = new List<double>();
+            foreach (var dbytes in bytesList)
+                doubleList.Add(BitConverter.ToDouble(dbytes, 0));
+            return doubleList.ToArray();
+        }
+    }
+
+    public decimal[] ToDecimalArray()
+    {
+        if (IsEmpty)
+            return new decimal[0];
+        else
+        {
+            try
+            {
+                var ints = ToInt32Array();
+                var intsList = new List<int[]>();
+                var c = -1;
+                for (int i = 0; i < ints.Length; i++)
+                {
+                    c++;
+                    if (i == 0 | c == 4)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)